### ISSUE TITLE
Improve Desired Nodes setting validation error messages

### ISSUE THREAD

**Issue by fcofdez** [elastic/elasticsearch#96464]
Currently the desired nodes API validates the node settings during updates, but the actual validation errors are presented as suppressed exceptions, those error messages are not returned by default in the REST API, making it unclear which setting is invalid. The API should return a clearer error message that tells users which settings are invalid.

**Comment by pxsalehi** [comment]
To me this is similar to [this issue](https://github.com/elastic/elasticsearch/issues/93468#issuecomment-1430986179), the Java code would see the nested exception (I think we often verify this in our unit tests), but in the REST API response, this is not visible. 

In this case, we have for example some tests in `DesiredNodesSettingsValidatorTests` to make sure that the correct error is included in the suppressed exceptions, but not in the REST response. This seems like a general problem, does someone know how do we usually change the REST error response in this cases instead of going with the default which seems to hide the original issue?

**Comment by pxsalehi** [comment]
Or is it rather changing `DesiredNodesSettingsValidator.validate` to return a better message that captures the message of the original exception?

**Comment by fcofdez** [comment]
> To me this is similar to https://github.com/elastic/elasticsearch/issues/93468#issuecomment-1430986179, the Java code would see the nested exception (I think we often verify this in our unit tests), but in the REST API response, this is not visible.

Yes, that's the issue here. The validation errors are saved as suppressed exceptions and those are not shown by default in the REST layer unless `error_trace` query parameter is `true`. I agree that this is a more general problem, but I'm not sure if there's an easy fix for this. My idea with this issue is to rework a bit how we present the top level error message.

**Comment by fcofdez** [comment]
> Would that be something like what I've mentioned in my last comment? Would just appending the first suppressed exception's message to the top level message help a bit at least?

That's an option, but it's possible that multiple settings from different nodes are invalid, so it's a bit tricky.

**Comment by pxsalehi** [comment]
We could limit it to one message per node on the top level. How did you imagine reworking the top level message?

**Comment by pxsalehi** [comment]
Another general way to mitigate this might be to at least append to the higher level error message something like "enable error_trace query parameter to see more detail" when there are suppressed exceptions. But that's a bit lazy and they'd have to repeat the request!

**Comment by fcofdez** [comment]
> We could limit it to one message per node on the top level. How did you imagine reworking the top level message?

I was thinking something like `Invalid settings in Nodes [<node_name>: [<setting_name>]...]` but I haven't thought deeply about this.

**Comment by pxsalehi** [comment]
Yes. I was thinking something very similar. Something like: `Nodes with ids [%s] in positions [%s] contain invalid settings: [<node_id>: error_message, ...]`. This would be an easy improvement I think. I'm not sure it is trivial to extract the setting name. I was assuming I can rely on the exception message coming from validating of that node to have a sensible message (including the setting name). It might be a reasonable expectation, since if that is not the case for some validation then we'd have to just improve that validator's message.

**Comment by fcofdez** [comment]
Yes, something like that would be an easy improvement 👍
